lab_3_generate_by_ngrams/main.py

```diff
@@ -331,6 +487,19 @@ def continue_sequence(
 
         In case of corrupt input arguments or unexpected behaviour of methods used return None.
         """
+        if not (isinstance(sequence, tuple) and sequence
+                and isinstance(next_tokens, list) and next_tokens
+                and isinstance(sequence_candidates, dict)
+                and sequence_candidates
+                and len(next_tokens) <= self._beam_width
+                and sequence in sequence_candidates):
+            return None
+        new_sequence_candidates = {}
+        for token_tuple in next_tokens:
+            new_sequence = sequence + (token_tuple[0],)
+            new_freq = sequence_candidates[sequence] - math.log(token_tuple[1])
+            new_sequence_candidates[new_sequence] = new_freq
+        return new_sequence_candidates
```

Comment:
This way you lose all the other sequences stored in `sequence_candidates` :<

Reply:
Fixed. I changed it and it passed one more test